File: 0367-valid-perfect-square/0367-valid-perfect-square.c
bool isPerfectSquare(int num) {
    if (num < 2)
        return true;

    int left = 1;
    int right = num / 2;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        long long sq = (long long)mid * mid;

        if (sq == num) {
            return true;
        }
        else if (sq < num) {
            left = mid + 1;
        }
        else {
            right = mid - 1;
        }
    }

    return false;
}

File: 0367-valid-perfect-square/0367-valid-perfect-square.py

class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num<2:
            return True
        l = 1
        r = num//2

        while l<=r:
            mid = (l+r)//2
            sq = mid*mid
            if sq == num:
                return True
            elif sq<num:
                l = mid+1
            else:
                r = mid-1
            
        return False

       
        

        